src/streams.h

```diff
@@ -69,6 +69,98 @@ OverrideStream<S> WithOrVersion(S* s, int nVersionFlag)
     return OverrideStream<S>(s, s->GetType(), s->GetVersion() | nVersionFlag);
 }
 
+/* Minimal stream for appending the contents of an existing byte vector
+ *
+ * The referenced vector will grow as necessary
+ */
+class CVectorInserter
+{
+ public:
+    CVectorInserter(int nTypeIn, int nVersionIn, std::vector<unsigned char>& vchDataIn) : nType(nTypeIn), nVersion(nVersionIn), vchData(vchDataIn)
+    {
+    }
+    template <typename... Args>
+    CVectorInserter(int nTypeIn, int nVersionIn, std::vector<unsigned char>& vchDataIn, Args&&... args) : CVectorInserter(nTypeIn, nVersionIn, vchDataIn)
+    {
+        ::SerializeMany(*this, std::forward<Args>(args)...);
+    }
+    template<typename T>
+    CVectorInserter& operator<<(const T& obj)
+    {
+        // Serialize to this stream
+        ::Serialize(*this, obj);
+        return (*this);
+    }
+    void write(const char* pch, size_t nSize)
+    {
+        vchData.insert(vchData.end(), reinterpret_cast<const unsigned char*>(pch), reinterpret_cast<const unsigned char*>(pch) + nSize);
+    }
+    int GetVersion() const
+    {
+        return nVersion;
+    }
+    int GetType() const
+    {
+        return nType;
+    }
+    void pad(size_t nSize)
+    {
+        vchData.insert(vchData.end(), nSize, 0);
+    }
+private:
+    const int nType;
+    const int nVersion;
+    std::vector<unsigned char>& vchData;
+};
+
+/* Minimal stream for overwriting the contents of an existing byte vector without causing reallocations
+ *
+ * Note that the referenced vector must already be large enough to hold all data to be serialized.
+ */
+class CVectorOverWriter
+{
+ public:
+    CVectorOverWriter(int nTypeIn, int nVersionIn, std::vector<unsigned char>& vchDataIn) : nType(nTypeIn), nVersion(nVersionIn), vchData(vchDataIn), nPos(0)
+    {
+    }
+    template <typename... Args>
+    CVectorOverWriter(int nTypeIn, int nVersionIn, std::vector<unsigned char>& vchDataIn, Args&&... args) : CVectorOverWriter(nTypeIn, nVersionIn, vchDataIn)
+    {
+        ::SerializeMany(*this, std::forward<Args>(args)...);
+    }
+    void write(const char* pch, size_t nSize)
+    {
+        assert(nSize + nPos <= vchData.size());
+        memcpy(vchData.data() + nPos, reinterpret_cast<const unsigned char*>(pch), nSize);
+        nPos += nSize;
+    }
+    template<typename T>
+    CVectorOverWriter& operator<<(const T& obj)
+    {
+        // Serialize to this stream
+        ::Serialize(*this, obj);
+        return (*this);
+    }
+    int GetVersion() const
+    {
+        return nVersion;
+    }
+    int GetType() const
+    {
+        return nType;
+    }
+    void skip(size_t nSize)
```

Comment:
CSizeComputer uses 'seek' for this function (inspired by the dd command, where skip is something you do on input, and seek is something you do on output).